src/clients/AxelarBridgeClient.ts

```diff
@@ -12,42 +12,71 @@ export interface DepositParams {
   senderAddress: string;
   receiverAddress: string;
   amount: BigNumber;
-  depositTokenExternalAddress: string;
+  depositTokenExternalAddress?: string;
   rpcUrl: string;
   gasPriceGwei?: BigNumber;
   gasLimit?: BigNumber;
   signer: ethers.Signer;
-  nonce?: number
+  nonce?: number;
+  isSupportNativeToken?: boolean;
 }
 
 export interface EthersTransactionResponse extends ethers.Transaction {
   wait: () => Promise<ethers.Transaction>;
 }
 
 export class AxelarBridgeClient {
-
-  private constructor(
-    public readonly configProvider: NetworkConfigProvider,
-  ) { }
+  private constructor(public readonly configProvider: NetworkConfigProvider) {}
 
   public static instance(opts: AxelarBridgeClientOpts) {
-    const { configProvider } = opts
-    return new AxelarBridgeClient(configProvider)
+    const { configProvider } = opts;
+    return new AxelarBridgeClient(configProvider);
   }
 
-  // lock deposit 
+  // lock deposit
   public async deposit(params: DepositParams): Promise<EthersTransactionResponse> {
-    const { contractAddress, senderAddress, receiverAddress, depositTokenExternalAddress, amount, signer, rpcUrl } = params;
-    const rpcProvider = new ethers.providers.JsonRpcProvider(rpcUrl)
-    const contract = new ethers.Contract(contractAddress, ABIs.axelarBridge, rpcProvider)
+    const {
+      contractAddress,
+      senderAddress,
+      receiverAddress,
+      depositTokenExternalAddress,
+      amount,
+      signer,
+      rpcUrl,
+      nonce,
+      gasPriceGwei,
+      gasLimit,
+      isSupportNativeToken = false,
+    } = params;
+    const rpcProvider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    const abi = isSupportNativeToken ? ABIs.nativeDepositer : ABIs.axelarBridge;
+    const contract = new ethers.Contract(contractAddress, abi, rpcProvider);
+
+    if (isSupportNativeToken) {
+      return await contract.connect(signer).deposit(
+        senderAddress, // tokenSender
+        receiverAddress, // carbonReceiver bech32Address
+        {
+          nonce,
+          ...(gasPriceGwei && { gasPrice: gasPriceGwei.shiftedBy(9).toString(10) }),
+          ...(gasLimit && { gasLimit: gasLimit.toString(10) }),
+          value: amount.toString(10),
+        }
```

Comment:
Think you can extract this object into a new variable and apply where necessary:
```typescript
const txParams = {
  nonce,
  ...(gasPriceGwei && { gasPrice: gasPriceGwei.shiftedBy(9).toString(10) }),
  ...(gasLimit && { gasLimit: gasLimit.toString(10) }),
  ...(isNativeTokenDeposit && { value: amount.toString(10) }),
}
...
return await contract.connect(signer).deposit(..., txParams);
```